src/amr/solvers/solver_ppc.hpp

```diff
@@ -218,373 +224,249 @@ void SolverPPC<HybridModel, AMR_Types>::restoreState_(level_t& level, Ions& ions
 
-        auto& Bpred = electromagPred_.B;
-        auto& B     = hybridState.electromag.B;
-        auto& Eavg  = electromagAvg_.E;
-
-        for (auto& patch : level)
-        {
-            auto _      = resourcesManager->setOnPatch(*patch, Bpred, B, Eavg);
-            auto layout = PHARE::amr::layoutFromPatch<GridLayout>(*patch);
-            auto __     = core::SetLayout(&layout, faraday_);
-            faraday_(B, Eavg, Bpred, dt);
-
-            resourcesManager->setTime(Bpred, *patch, newTime);
-        }
+        for (auto& state : view)
+            resourcesManager->setTime(state.electromagPred.B, *state.patch, newTime);
     }
 
-
     {
         PHARE_LOG_SCOPE("SolverPPC::predictor2_.ampere");
-
-        auto& Bpred = electromagPred_.B;
-        auto& J     = hybridState.J;
-
-
-        for (auto& patch : level)
-        {
-            auto _      = resourcesManager->setOnPatch(*patch, Bpred, J);
-            auto layout = PHARE::amr::layoutFromPatch<GridLayout>(*patch);
-            auto __     = core::SetLayout(&layout, ampere_);
-            ampere_(Bpred, J);
-
-            resourcesManager->setTime(J, *patch, newTime);
-        }
-        fromCoarser.fillCurrentGhosts(J, levelNumber, newTime);
+        ampere_(view.layouts, view.electromagPred_B, view.J);
+        for (auto& state : view)
+            resourcesManager->setTime(state.J, *state.patch, newTime);
+        fromCoarser.fillCurrentGhosts(view.model().state.J, level.getLevelNumber(), newTime);
     }
 
-
     {
         PHARE_LOG_SCOPE("SolverPPC::predictor2_.ohm");
-
-        auto& electrons = hybridState.electrons;
-        auto& Bpred     = electromagPred_.B;
-        auto& Epred     = electromagPred_.E;
-        auto& J         = hybridState.J;
-
-        for (auto& patch : level)
-        {
-            auto layout = PHARE::amr::layoutFromPatch<GridLayout>(*patch);
-            auto _      = resourcesManager->setOnPatch(*patch, Bpred, Epred, J, electrons);
-            electrons.update(layout);
-            auto& Ve = electrons.velocity();
-            auto& Ne = electrons.density();
-            auto& Pe = electrons.pressure();
-            auto __  = core::SetLayout(&layout, ohm_);
-            ohm_(Ne, Ve, Pe, Bpred, J, Epred);
-            resourcesManager->setTime(Epred, *patch, newTime);
-        }
+        for (auto& state : view)
+            state.electrons.update(state.layout);
+        ohm_(view.layouts, view.electrons_N, view.electrons_Ve, view.electrons_Pe,
+             view.electromagPred_B, view.J, view.electromagPred_E);
+        for (auto& state : view)
+            resourcesManager->setTime(state.electromagPred.E, *state.patch, newTime);
     }
 }
 
 
 
 
 template<typename HybridModel, typename AMR_Types>
-void SolverPPC<HybridModel, AMR_Types>::corrector_(level_t& level, HybridModel& model,
+void SolverPPC<HybridModel, AMR_Types>::corrector_(level_t& level, HybridModelView_t& view,
```

Comment:
the full name of the implementation is `HybridPPCModelView ` 

in this case it could be called just ModelView, as teh solver ppc part is implicit somewhat